src/test/java/org/folio/notes/controller/NoteTypesControllerTest.java

```diff
@@ -0,0 +1,347 @@
+    );
+    databaseHelper.saveNoteTypes(noteTypes, TENANT);
+
+    var cqlQuery = "(cql.allRecords=1)sortby name/sort.descending";
+    var limit = "1";
+    var offset = "1";
+    mockMvc.perform(get(BASE_URL + "?limit={l}&offset={o}&query={cql}", limit, offset, cqlQuery)
+        .headers(defaultHeaders()))
+      .andExpect(matchAll(
+        status().isOk(),
+        jsonPath("$.noteTypes.[0].name", is(noteTypes.get(1).getName())),
+        jsonPath("$.noteTypes.[1]").doesNotExist(),
+        jsonPath("$.totalRecords").value(3)
+      ));
+  }
+
+  @Test
+  @DisplayName("Find all note-types by name")
+  void returnCollectionByName() throws Exception {
+    List<NoteTypeEntity> noteTypes = List.of(
+      NoteTypeEntity.builder().name("first").build(),
+      NoteTypeEntity.builder().name("second").build(),
+      NoteTypeEntity.builder().name("third").build()
+    );
+    databaseHelper.saveNoteTypes(noteTypes, TENANT);
+
+    var cqlQuery = "name=third";
+    mockMvc.perform(get(BASE_URL + "?query={cql}", cqlQuery).headers(defaultHeaders()))
+      .andExpect(matchAll(
+        status().isOk(),
+        jsonPath("$.noteTypes.[0].name", is(noteTypes.get(2).getName())),
+        jsonPath("$.noteTypes.[1]").doesNotExist(),
+        jsonPath("$.totalRecords").value(1)
+      ));
+  }
+
+  @Test
+  @DisplayName("Return 422 on get collection with invalid CQL query")
+  void return422OnGetCollectionWithInvalidCqlQuery() throws Exception {
+    var cqlQuery = "!invalid-cql!";
+    mockMvc.perform(get(BASE_URL + "?query={cql}", cqlQuery)
+        .headers(defaultHeaders()))
+      .andExpect(status().isUnprocessableEntity())
+      .andExpect(exceptionMatch(CqlQueryValidationException.class))
+      .andExpect(errorMessageMatch(containsString("Not implemented yet node type")));
+  }
+
+  @Test
+  @DisplayName("Return 422 on get collection with invalid offset")
+  void return422OnGetCollectionWithInvalidOffset() throws Exception {
+    mockMvc.perform(get(BASE_URL + "?offset={offset}", -1)
+        .headers(defaultHeaders()))
+      .andExpect(status().isUnprocessableEntity())
+      .andExpect(exceptionMatch(ConstraintViolationException.class))
+      .andExpect(errorMessageMatch(containsString("must be greater than or equal to 0")));
+  }
+
+  @Test
+  @DisplayName("Return 422 on get collection with invalid limit")
+  void return422OnGetCollectionWithInvalidLimit() throws Exception {
+    mockMvc.perform(get(BASE_URL + "?limit={limit}", -1)
+        .headers(defaultHeaders()))
+      .andExpect(status().isUnprocessableEntity())
+      .andExpect(exceptionMatch(ConstraintViolationException.class))
+      .andExpect(errorMessageMatch(containsString("must be greater than or equal to 1")));
+  }
+
+  // Tests for POST
+
+  @Test
+  @DisplayName("Create new note-type")
+  void createNewNoteType() throws Exception {
+    String name = "First";
+    NoteType noteType = new NoteType().name(name);
+    noteType.setId(UUID.randomUUID());
```

Comment:
Please add the same test but without an id and check that it was populated

Reply:
it's not, hibernate sad that
`ids for this class must be manually assigned before calling save(): org.folio.notes.domain.entity.NoteTypeEntity`